### PR TITLE
Loki: Add format explanation to regex operations

### LINKED ISSUE #59221: Loki Query Builder: add format explanation for regular expression filter
# Context

While testing the Query Builder, we noticed that there is no clue about the expected regular expression format for this filter. For example, the user attempted to use a JS regular expression, and the correct format is RE2 regex: https://github.com/google/re2/wiki/Syntax

imagen

# Expectations
- Add a reference of the expected format to use.
  imagen

### PR DESCRIPTION
**Pull request by itsgareth**
**What is this feature?**
adds an explanation to what format should be used when using regex operations 

**Why do we need this feature?**
right now it is not clear what regex format should be used.
users are trying JavaScript regex (`/error/i`) when the correct format is Go  (`(?i)error`)

**Who is this feature for?**
...

**Which issue(s) does this PR fix?**:
Fixes #59221

**Special notes for your reviewer**:
...

### CODE REVIEW

File: public/app/plugins/datasource/loki/querybuilder/operations.ts [hunk 1]
```diff
@@ -324,7 +324,7 @@ Example: \`\`error_level=\`level\` \`\`
       orderRank: LokiOperationOrder.LineFilters,
       renderer: getLineFilterRenderer('|~'),
       addOperationHandler: addLokiOperation,
-      explainHandler: (op) => `Return log lines that match regex \`${op.params[0]}\`.`,
+      explainHandler: (op) => `Return log lines that match a \`Go language\` regex pattern. \`${op.params[0]}\`.`,
```
Comment (svennergr) [inline public/app/plugins/datasource/loki/querybuilder/operations.ts:327]: I think we should come up with something better than `Go language`. The RegExp syntax is not really tied to go, but is used there a lot.
Could we just name it `RE2 syntax` or something along the lines of that?

https://github.com/google/re2/wiki/Syntax

Comment (gwdawson) [reply public/app/plugins/datasource/loki/querybuilder/operations.ts:327]: of course, thats my bad i didn't know that. i'll change it now

Comment (matyax) [reply public/app/plugins/datasource/loki/querybuilder/operations.ts:327]: It was our fault. We specified it like that in the original issue. Thanks both!

**Review by matyax — APPROVED** [review]
Looks good!

**Pull request merged**